src/store.js

```diff
@@ -16,40 +21,14 @@ export const missingStore = () => (
   `)
 );
 
-// Used for querying the store #where / #find
-const deepPartialEqual = (obj1, obj2) => (
-  Object.entries(obj1).reduce((equal, [key, value]) => {
-    if (equal === false) return equal;
-    if (obj2 === undefined) return false;
+const toModel = (Model) => (data = null) => (data ? new Model(data) : data);
+const mapToModel = (Model) => (data) => data.map((item) => toModel(Model)(item));
 
-    if (typeof value === 'object' && !Array.isArray(value)) {
-      return deepPartialEqual(value, obj2[key]);
-    }
-
-    if (value !== obj2[key]) return false;
-
-    return equal;
-  }, true)
-);
-
-// Query through an array of objects
-export const queryData = (entities, query) => (
-  entities.filter((entity) => (
-    Object.entries(query).reduce((match, [queryKey, queryValue]) => {
-      if (match === false) return match;
-
-      const entityValue = entity[queryKey];
-
-      if (Array.isArray(queryValue)) {
-        return queryValue.includes(entityValue);
-      }
-      if (typeof queryValue === 'object') {
-        return deepPartialEqual(queryValue, entityValue);
-      }
-      return queryValue === entityValue;
-    }, true)
-  ))
-);
+const createRecordPromise = ({ promise, selector, store }) => {
```

Comment:
Same thing here about subclassing primitives - I could subclass `Promise` and wrap it, but I think my intentions are fairly clear.